_topics/en/time-warp.md

```diff
@@ -0,0 +1,159 @@
+  prevent difficulty from increasing even as the rate of block
+  production increases.
+
+---
+Ignoring some technical details, Bitcoin's difficulty adjustment
+algorithm (DAA) attempts to keep an average of ten minutes between
+new blocks by adjusting the difficulty of the proof of work (PoW) that
+blocks must demonstrate in order to be valid.  This depends on miners
+including the (approximate) current time in the block headers they
+create, so that the Bitcoin consensus protocol can calculate how much the
+recent inter-block average differs from the ideal of ten minutes.
+
+Consensus rules prevent a single miner, or small number of miners, from
+including manipulative times in their block headers by requiring that
+any time in a block header must be greater than the median time of the
+previous 11 blocks, called the _median time past_ (MTP) rule.
+
+However, miners controlling a majority of hashrate (or a sub-majority of
+miners using an attack that gives them influence over other miners, such
+as _selfish mining_), can gain precise control over the header time of
+each block.  For almost all blocks, they can set this to the minimum
+increase of one second greater than the median.  But, for the last block
+in a difficulty adjustment period, they can set it to the current time.
+Using small numbers for illustrative purposes below, we can see a series
+of header times suddenly advance far in the apparent future and then
+return to the past, giving the attack its _time warp_ name:
+
+    [0, 1, 2, 1795594, 3, 4, 5, ...]
+
+Bitcoins DAA only adjusts difficulty once every 2016 blocks.  Miners who
+begin a time warp attack cannot significantly affect difficulty for the
+first 2016-block epoch because both the first and last blocks in that
+epoch will need to have approximately accurate times.  For the second
+epoch, miners can make it appear that the first block was mined two
+weeks in the past and the last block was mined at present, giving an
+average of about 20 minutes per block and causing the DAA to halve the
+difficulty.  Miners will then by able to complete the next epoch twice
+as fast and yet make it seem like it took about 25 minutes per block,
+lowering difficulty further.  They can repeat the attack indefinitely
+until they're producing one block per second, the lower bound allowed by
+the MTP rule.
+
+## Consequences
+
+The original reason to fear the attack was that miners could use it to
+quickly claim all remaining block subsidy.  The attack takes about a
+month of publicly visible preparation before it really gets going, at
+which point it would only take miners about another month to claim all
+remaining subsidy, which was valued at approximately $91 billion USD at
+the time of this writing (April 2024).
+
+Since the discovery of the attack, a significant amount of user funds
+(but believed to be far less than $91 billion) is new stored in contract
+protocols that use [timelocks][topic timelocks].  Since the activation of
```

Comment:
```suggestion
(but believed to be far less than $91 billion) is now stored in contract
protocols that use [timelocks][topic timelocks].  Since the activation of
```